examples/routing_service/routing_service_msql_adapter/cpp/rs_main/MainRsMsqlShapesAdapter_w_types.xml

```diff
@@ -0,0 +1,376 @@
+<?xml version="1.0"?>
+<!--
+  (c) 2020 Copyright, Real-Time Innovations, Inc.  All rights reserved.
+  
+  RTI grants Licensee a license to use, modify, compile, and create derivative
+  works of the Software.  Licensee has the right to distribute object form
+  only for use with RTI products.  The Software is provided "as is", with no
+  warranty of any type, including any warranty for fitness for any purpose.
+  RTI is under no obligation to maintain or support the Software.  RTI shall
+  not be liable for any incidental or consequential damages arising out of the
+  use or inability to use the software.
+-->
+<dds xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://community.rti.com/schema/6.0.1/rti_routing_service.xsd">
+
+  <qos_library name="RsMsqlQosLib">
+    <qos_profile name="RsMsqlQosReliableProfile" base_name="BuiltinQosLibExp::Generic.KeepLastReliable">
+      <datareader_qos>
+        <history>
+          <kind>KEEP_LAST_HISTORY_QOS</kind>
+        </history>
+      </datareader_qos>
+      <datawriter_qos>
+        <history>
+          <kind>KEEP_LAST_HISTORY_QOS</kind>
+        </history>
+        <lifespan>
+          <duration>
+            <sec>5</sec>
+            <nanosec>0</nanosec>
+          </duration>
+        </lifespan>
+      </datawriter_qos>
+    </qos_profile>
+  </qos_library>
+
+  <types>
```

Comment:
Maybe you could have a single file and avoid registering the types you want to autodiscover. Having 4 XML files that do almost the same is a bit confusing.

Reply:
I would like to leave it as is. My blog explains each XML file.